cobc/ppparse.y

```diff
@@ -1145,32 +1138,18 @@ refmod_directive:
 ;
 
 source_directive:
-  _format _is format_type
+  _format _is VARIABLE_NAME
   {
+	  if (!cobc_deciph_source_format ($3)) {
+		  ppp_error_invalid_option ("SOURCE", $3);
+	  }
 	  if (cb_src_list_file) {
-		  cb_current_file->source_format = cb_source_format;
+		  cb_current_file->source_format = cobc_get_source_format ();
 	  }
   }
-;
-
-format_type:
-  FIXED
-  {
-	cb_source_format = CB_FORMAT_FIXED;
-	cb_text_column = cb_config_text_column;
-  }
-| FREE
-  {
-	cb_source_format = CB_FORMAT_FREE;
-  }
-| VARIABLE
-  {
-	cb_source_format = CB_FORMAT_FIXED;
-	cb_text_column = 500;
-  }
-| GARBAGE
+| _format _is GARBAGE
```

Comment:
Do we still get in the GARBAGE path with the VARIABLE_NAME token being based on WORD?

Reply:
Actually we can get a LITERAL; I'll fix this.